GlobalstatsIO.cs

```diff
@@ -5,318 +5,315 @@
+
+            if (www.isNetworkError || www.isHttpError)
+            {
+                Debug.LogWarning("Error retrieving access token: " + www.error);
+                Debug.Log("GlobalstatsIO API Response: " + responseBody);
+                yield break;
+            }
+            else
+            {
+                api_access_token = JsonUtility.FromJson<AccessToken>(responseBody);
+            }
+        }
+
+        public IEnumerator Share(string id = "", string name = "", Dictionary<string, string> values = null, Action<bool> callback = null)
+        {
+            bool update = false;
+
+            if (api_access_token == null || !api_access_token.IsValid())
+            {
+                yield return GetAccessToken();
+            }
+
+            // If no id is supplied but we have one stored, reuse it.
+            if (id == "" && statistic_id != "")
+                id = statistic_id;
+
+            string url = "https://api.globalstats.io/v1/statistics";
+            if (id != "")
+            {
+                url = "https://api.globalstats.io/v1/statistics/" + id;
+                update = true;
+            }
+            else
+            {
+                if (name == "")
+                    name = "anonymous";
+            }
+
+            string json_payload;
+
+            if (update == false)
+            {
+                json_payload = "{\"name\":\"" + name + "\", \"values\":{";
+            }
+            else
+            {
+                json_payload = "{\"values\":{";
+            }
+
+            bool semicolon = false;
+            foreach (KeyValuePair<string, string> value in values)
+            {
+                if (semicolon)
+                    json_payload += ",";
+                json_payload += "\"" + value.Key + "\":\"" + value.Value + "\"";
+                semicolon = true;
+            }
+            json_payload += "}}";
+
+            byte[] pData = Encoding.UTF8.GetBytes(json_payload);
+            UnityWebRequest www;
+
+            if (update == false)
+            {
+                www = new UnityWebRequest(url, "POST")
+                {
+                    uploadHandler = new UploadHandlerRaw(pData)
+                };
+            }
+            else
+                www = UnityWebRequest.Put(url, pData);
+
+            www.downloadHandler = new DownloadHandlerBuffer();
+            www.SetRequestHeader("Authorization", "Bearer " + api_access_token.access_token);
+            www.SetRequestHeader("Content-Type", "application/json");
+            yield return www.SendWebRequest();
+
+            string responseBody = www.downloadHandler.text;
+
+            StatisticResponse statistic = null;
+
+            if (www.isNetworkError || www.isHttpError)
+            {
+                Debug.LogWarning("Error submitting statistic: " + www.error);
+                Debug.Log("GlobalstatsIO API Response: " + responseBody);
+                callback(false);
```

Comment:
You've made `callback` optional: `Action<bool> callback = null`.
You can use:  `callback?.Invoke(false);` so that it doesn't throw when `callback == null`.